cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -343,14 +361,32 @@ func translateUpstreamNameServerOfKubeDNSToUpstreamProxyCoreDNS(dataField string
 	}
 
 	if upstreamValues, ok := kubeDNSConfigMap.Data[dataField]; ok {
-		var upstreamProxyIP []string
+		var upstreamProxyValues []string
 
-		err := json.Unmarshal([]byte(upstreamValues), &upstreamProxyIP)
+		err := json.Unmarshal([]byte(upstreamValues), &upstreamProxyValues)
 		if err != nil {
 			return "", errors.Wrap(err, "failed to parse JSON from 'kube-dns ConfigMap")
 		}
+		// Check if the data extracted from the kube-dns ConfigMap contains a valid
+		// IP address. Hostname to nameservers is not supported on CoreDNS and omit the
+		// hostname, if there are any present.
+		index := -1
+		for i, value := range upstreamProxyValues {
+			upstreamProxyHost, _, err := kubeadmutil.ParseHostPort(value)
+			if err != nil {
+				return "", errors.Wrap(err, "invalid format to parse for upstreamnameserver")
+			}
+			if net.ParseIP(upstreamProxyHost) == nil {
+				index = i
+				fmt.Printf("[WARNING]: Your upstreamnameserver Config contains a hostname %v, it will be omitted from the translation.\n", upstreamProxyHost)
+			}
+		}
+		if index != -1 {
```

Comment:
Unrelated, but shouldn't this function be called ....

`translateUpstreamNameServerOfKubeDNSToForwardCoreDNS` instead of ...

`translateUpstreamNameServerOfKubeDNSToUpstreamProxyCoreDNS`